Firmware/mmu2_log.h

```diff
@@ -20,18 +20,36 @@ void LogEchoEvent_P(const char *msg);
 
 } // namespace
 
-#define SERIAL_MMU2() { serialprintPGM(mmu2Magic); }
-
-#define MMU2_ECHO_MSGLN(S) do{ SERIAL_ECHO_START; SERIAL_MMU2(); SERIAL_ECHOLN(S); }while(0)
-#define MMU2_ERROR_MSGLN(S) MMU2_ECHO_MSGLN(S) //!@todo Decide MMU2 errors  on serial line
-#define MMU2_ECHO_MSGRPGM(S) do{ SERIAL_ECHO_START; SERIAL_MMU2(); SERIAL_ECHORPGM(S); }while(0)
-#define MMU2_ERROR_MSGRPGM(S) MMU2_ECHO_MSGRPGM(S) //!@todo Decide MMU2 errors  on serial line
+    #define SERIAL_MMU2() \
+        { serialprintPGM(mmu2Magic); }
+
+    #define MMU2_ECHO_MSGLN(S)   \
+        do {                     \
+            SERIAL_ECHO_START;   \
+            SERIAL_MMU2();       \
+            SERIAL_ECHOLN(S);    \
+        } while (0)
+    #define MMU2_ERROR_MSGLN(S) MMU2_ECHO_MSGLN(S) //!@todo Decide MMU2 errors  on serial line
+    #define MMU2_ECHO_MSGRPGM(S) \
+        do {                     \
+            SERIAL_ECHO_START;   \
+            SERIAL_MMU2();       \
+            SERIAL_ECHO(S);      \
```

Comment:
⚠️ This must be `SERIAL_ECHORPGM(S)` for PGM strings, otherwise the serial message is corrupted. This negatively impacts the printer and it appears to run more slowly due the corruption seems to "spam" the serial stream.

Reply:
Fixed in [f79dfac](https://github.com/prusa3d/Prusa-Firmware/pull/3950/commits/f79dfac133e99d4335e6aaa3eef6c4e937db7a45)